### PR TITLE
[Merged by Bors] - feat(Topology/UniformSpace): `Subtype.isComplete_iff`

### PR DESCRIPTION
**Pull request by pechersky**
In the style of `Subtype.isCompact_iff`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 1]
```diff
@@ -265,6 +265,21 @@ theorem isComplete_image_iff {m : α → β} {s : Set α} (hm : UniformInducing
   simp_rw [IsComplete, imp.swap (a := Cauchy _), ← mem_Iic (b := 𝓟 _), fact1.forall fact2,
     hm.cauchy_map_iff, exists_mem_image, map_le_iff_le_comap, hm.inducing.nhds_eq_comap]
 
+/-- If `f : X → Y` is an `Inducing` map, the image `f '' s` of a set `s` is complete
+  if and only if `s` is complete. -/
```
Comment (urkud) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:269]: You say `Inducing`, should be `UniformInducing`. Also, LHS/RHS disagree with the lemma (see also below).

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 2]
```diff
@@ -265,6 +265,21 @@ theorem isComplete_image_iff {m : α → β} {s : Set α} (hm : UniformInducing
...
+theorem UniformInducing.isComplete_iff {f : α → β} {s : Set α} (hf : UniformInducing f) :
+    IsComplete s ↔ IsComplete (f '' s) := (isComplete_image_iff hf).symm
```
Comment (urkud) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:271]: I'm a bit worried that the LHS looks "simpler" than the RHS. Should we just add an alias for `isComplete_image_iff` (or move it to the `UniformInducing` namespace)?

Comment (pechersky) [reply Mathlib/Topology/UniformSpace/UniformEmbedding.lean]: I fixed up the two "simpler LHS" lemmas, leaving the `Subtype` one the way it is. Should I also flip the `Inducting.isCompact_iff` parallel lemmas?

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean [hunk 3]
```diff
@@ -265,6 +265,21 @@ theorem isComplete_image_iff {m : α → β} {s : Set α} (hm : UniformInducing
...
+  if and only if `s` is complete. -/
+theorem UniformInducing.isComplete_iff {f : α → β} {s : Set α} (hf : UniformInducing f) :
+    IsComplete (f '' s) ↔ IsComplete s := isComplete_image_iff hf
+
+/-- If `f : X → Y` is an `UniformEmbedding`, the image `f '' s` of a set `s` is complete
+  if and only if `s` is complete. -/
+theorem UniformEmbedding.isComplete_iff {f : α → β} {s : Set α} (hf : UniformEmbedding f) :
+    IsComplete (f '' s) ↔ IsComplete s := hf.toUniformInducing.isComplete_iff
+
+/-- Sets of subtype are complete iff the image under a coercion is. -/
```
Comment (b-mehta) [inline Mathlib/Topology/UniformSpace/UniformEmbedding.lean:278]:
```suggestion
/-- Sets of a subtype are complete iff their image under the coercion is complete. -/
```
This reads better to me, what do you think?

**Review by b-mehta — APPROVED** [review]

File: Mathlib/Topology/UniformSpace/UniformEmbedding.lean (hunk 2 above)
Comment (j-loreaux) [reply Mathlib/Topology/UniformSpace/UniformEmbedding.lean]: I'm not too worried about it because they aren't `@[simp]` lemmas

**Pull request closed without merging**